### PR TITLE
Add support for Lenovo A6010

### PR DESCRIPTION
**Pull request by jenneron**

### CODE REVIEW

File: arch/arm64/boot/dts/qcom/msm8916-lenovo-a6010.dts [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+	};
+
+	l10 {
+		regulator-min-microvolt = <2700000>;
+		regulator-max-microvolt = <2800000>;
+	};
+
+	l11 {
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <3300000>;
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-lenovo-a6010.dts:175]: This seems to be fake, the RPM firmware suggests this is actually limited to 2.95V (which was the default value here). I would change it back to 2.95V.

**Review by Minecrell — CHANGES_REQUESTED** [review]

File: arch/arm64/boot/dts/qcom/Makefile [hunk 2]
```diff
@@ -7,6 +7,7 @@ dtb-$(CONFIG_ARCH_QCOM)	+= apq8096-ifc6640.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= ipq6018-cp01-c1.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= ipq8074-hk01.dtb
 dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-bq-paella.dtb msm8916-bq-paella-modem.dtb
+dtb-$(CONFIG_ARCH_QCOM)	+= msm8916-lenovo-a6010.dtb
```
[the remark this answers is not in the document]
Comment (jenneron) [reply arch/arm64/boot/dts/qcom/Makefile]: i appended it before wt88047

File: arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86528.dts [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+#include "msm8916-pm8916.dtsi"
+#include "msm8916-no-psci.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/leds/common.h>
+#include <dt-bindings/pinctrl/qcom,pmic-mpp.h>
+
+/ {
+	model = "Lenovo A6010 (Wingtech WT86528)";
+	compatible = "lenovo,a6010", "qcom,msm8916";
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86528.dts:13]:
```suggestion
	compatible = "wingtech,wt86528", "qcom,msm8916";
```

We could list `lenovo,a6010` additionally but not sure if that is worth it. This compatible isn't used anywhere.

**Review by Minecrell — APPROVED** [review]

File: arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86528.dts [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&gpio_leds_default>;
+
+		red {
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-wingtech-wt86528.dts:44]: Apparently we are supposed to call these `led-0`, `led-1` etc:

https://github.com/msm8916-mainline/linux/blob/48cc8d930cb544b00cbdcf288ee17a02912d34fd/Documentation/devicetree/bindings/leds/leds-gpio.yaml#L55-L69

**Review by Minecrell — CHANGES_REQUESTED** [review]

**Review by Minecrell — APPROVED** [review]

**Comment by Minecrell** [comment]
Pushed, thanks!

**Pull request closed without merging**